app/livechat/server/api/lib/visitors.js

```diff
@@ -62,13 +63,40 @@ export async function findChatHistory({ userId, roomId, visitorId, pagination: {
 	});
 
 	const total = await cursor.count();
-
 	const history = await cursor.toArray();
-
-	return {
-		history,
-		count: history.length,
-		offset,
-		total,
-	};
+	if(text == 'null'){
+		return {
+			history,
+			count: history.length,
+			offset,
+			total,
+			
+		};
+		
+	}else{
+		let resultArray=[];
+		Meteor.runAsUser(userId,()=>{
+			for(var i=0; i<history.length; i++){
+				var roomid = history[i]._id;
+				var count = 1000;
```

Comment:
Why 1000?
You just one match to know that the room has at least one message with the search term.

Reply:
okay now count is 1